test/jasmine/tests/sankey_test.js

```diff
@@ -610,27 +666,36 @@ describe('sankey tests', function() {
             .then(done);
         });
 
-        it('should not output hover/unhover event data when hovermoder is false', function(done) {
-            var fig = Lib.extendDeep({}, mock);
-
-            Plotly.plot(gd, fig)
-            .then(function() { return Plotly.relayout(gd, 'hovermode', false); })
-            .then(function() { return _hover('node'); })
-            .then(failTest).catch(function(err) {
-                expect(err).toBe('plotly_hover did not get called!');
-            })
-            .then(function() { return _unhover('node'); })
-            .then(failTest).catch(function(err) {
-                expect(err).toBe('plotly_unhover did not get called!');
-            })
-            .then(function() { return _hover('link'); })
+        function assertNoHoverEvents(type) {
+            return Promise.resolve()
+            .then(function() { return _hover(type); })
             .then(failTest).catch(function(err) {
                 expect(err).toBe('plotly_hover did not get called!');
             })
-            .then(function() { return _unhover('link'); })
+            .then(function() { return _unhover(type); })
             .then(failTest).catch(function(err) {
                 expect(err).toBe('plotly_unhover did not get called!');
-            })
+            });
+        }
+
+        it('should not output hover/unhover event data when hovermoder is false', function(done) {
+            var fig = Lib.extendDeep({}, mock);
+
+            Plotly.plot(gd, fig)
+            .then(function() { return Plotly.relayout(gd, 'hovermode', false); })
+            .then(function() { return assertNoHoverEvents('node');})
+            .then(function() { return assertNoHoverEvents('link');})
```

Comment:
Nonblocking, but for extra style points you can move the function wrapper into `assertNoHoverEvents` so this becomes just: `.then(assertNoHoverEvents('link'))`

Reply:
Good idea! Fixed in 706e7ae